pkg/operator/ceph/cluster/osd/create.go

```diff
@@ -381,5 +398,54 @@ func createDaemonOnNode(c *Cluster, osd OSDInfo, nodeName string, config *provis
 	updateConditionFunc(c.clusterInfo.Context, c.context, c.clusterInfo.NamespacedName(), k8sutil.ObservedGenerationNotAvailable, cephv1.ConditionProgressing, v1.ConditionTrue, cephv1.ClusterProgressingReason, message)
 
 	_, err = k8sutil.CreateDeployment(c.clusterInfo.Context, c.context.Clientset, d)
-	return errors.Wrapf(err, "failed to create deployment for OSD %d on node %q", osd.ID, nodeName)
+	if err != nil {
+		return errors.Wrapf(err, "failed to create deployment for OSD %d on node %q", osd.ID, nodeName)
+	}
+
+	if c.spec.Network.MultiClusterService.Enabled {
+		updatedDeployment, err := updateDeploymentArgs(c, osd, d)
+		if err != nil {
+			return errors.Wrapf(err, "failed to update external IP in deployment for OSD %d on node %q", osd.ID, nodeName)
+		}
+
+		_, err = k8sutil.CreateOrUpdateDeployment(c.clusterInfo.Context, c.context.Clientset, updatedDeployment)
+		if err != nil {
+			return errors.Wrapf(err, "failed to update osd %d deployment args with external IP", osd.ID)
+		}
+	}
+
+	return nil
+}
+
+func updateDeploymentArgs(c *Cluster, osd OSDInfo, deployment *appsv1.Deployment) (*appsv1.Deployment, error) {
```

Comment:
How about we split this method into two steps? Create the service, and then update the OSD pod spec. Then it will be easier to unit test.